### ISSUE TITLE
TODO: Fix crash on Windows

### ISSUE THREAD

**Issue by bangerth** [bangerth/biharmonic#9]
Riccardo reports the following when running on Windows:
```
--------------------------------------------------------
An error occurred in line <2886> of file <C:\Users\riccardo\source\dealii\include\deal.II/lac/affine_constraints.templates.h> in function
    ScratchDataAccessor
The violated condition was:
    my_scratch_data->in_use == false
Additional information:
    Access to thread-local scratch data tried, but it is already in use
--------------------------------------------------------
```
This clearly needs to be fixed.

**Comment by bangerth** [comment]
The root cause is this one: https://github.com/dealii/dealii/issues/10284

**Comment by bangerth** [comment]
Work-around implemented in b804068a165487cea13b2cd525f0426a128722b9.
